### ISSUE TITLE
Adding service_id parameter does not work

### ISSUE THREAD

**Issue by darrellnorton** [3scale/3scale_ws_api_for_dotnet#5]
If I need to submit the service_id parameter to the report method, it throws an exception. 

I added the following to correct the issue in the AddTransactions private method in Api.cs:

```
        foreach (var entry in transactions.Keys)
        {
            // Added this to correct an issue submitting service_id (valid parameter per 3scale's API)
            if (entry.Equals("service_id"))
            {
                content += string.Format("&{0}={1}", entry, transactions[entry]);
            }
            else
            {
                Hashtable transaction = (Hashtable)transactions[entry];
```

**Comment by jrvariya** [comment]
Hi darrellnorton, 
I have faced the same issue while record usage after oauth request.
I have updated AddTransactions Function code as below, its generic check for type of key value pair ,hope so it will help to resolve your issue.
` private void AddTransactions(ref string content, Hashtable transactions)
        {
            foreach (var entry in transactions.Keys)
            {
                if (transactions[entry] is string)
                {
                    content += string.Format("&{0}={1}", entry, transactions[entry]);
                }
                else
                {
                    Hashtable transaction = (Hashtable)transactions[entry];
                    foreach (string parameter in transaction.Keys)
                    {
                        if (parameter.Equals("usage"))
                        {
                            Hashtable usage = (Hashtable)transaction[parameter];

                            if ((usage == null) || (usage.Count <= 0))
                                throw new ApiException("argument error: undefined transaction, usage is missing in one record");

                            foreach (string metric in usage.Keys)
                            {
                                content += string.Format("&transactions[{0}][{1}][{2}]={3}", entry, parameter, metric, usage[metric]);
                            }
                        }
                        else
                        {
                            content += string.Format("&transactions[{0}][{1}]={2}", entry, parameter, transaction[parameter]);
                        }
                    }
                }
            }
        }`

**Comment by mpguerra** [comment]
This should be fixed in 2.1.0, the way to specify the service_id parameter for reports would be: 

```c#
// Try report
                System.Collections.Hashtable transactions = new System.Collections.Hashtable();
                System.Collections.Hashtable transaction = null;
                transaction = new System.Collections.Hashtable();
                transaction.Add("app_id", app_id);
                transaction.Add("service_id", service_id);
                //transaction.Add("user_key", user_key);
                usage = new System.Collections.Hashtable();
                usage.Add("hits", 10);
                transaction.Add("usage", usage);
                transactions.Add("0", transaction);

                transaction = new System.Collections.Hashtable();
                transaction.Add("app_id", app_id);
                //transaction.Add("user_key", user_key);
                usage = new System.Collections.Hashtable();
                usage.Add("hits", 1);
                transaction.Add("usage", usage);
                transactions.Add("1", transaction);
                
                _3ScaleAPI.report(transactions);
```

See [ConsoleDemo](3scale/3scale_ws_api_for_dotnet/blob/master/src/ConsoleDemo/Program.cs)  for more details
